### ISSUE TITLE
Debugger click-to-resume overlay broken on 0.19

### ISSUE THREAD

**Issue by klazuka** [elm/browser#16]
There's a regression in the Elm 0.19 debugger's "click-to-resume" feature. In Elm 0.18, when you time-travel into the past, a big "click-to-resume" overlay covers your app and clicking it time travels to the present. But in 0.19, the overlay is *not shown*. Clicking anywhere in your app while in this state will do nothing. It's especially bad if the user closes the debugger popup window while in this state, as they will be unable to reopen the debugger in order to resume.

## Steps to reproduce

First, compile `Main.elm` using `--debug`
```elm
module Main exposing (..)

import Browser
import Html exposing (button, div, text)
import Html.Attributes exposing (id)
import Html.Events exposing (onClick)

main = Browser.sandbox { init = 0 , view = view , update = update }

type Msg = Increment

update _ n = n + 1

view n =
    div []
        [ text (String.fromInt n)
        , button [ onClick Increment ] [ text "+" ]
        ]
```
1. open the resulting `index.html`
1. click the "+" button a few times
1. open the Elm debugger popup
1. select a msg from the past
1. observe that inputs are blocked, but no overlay is shown

Here's a [video](https://www.dropbox.com/s/8h70b42zatdv2ou/broken%20debugger%20example.mov?dl=0) demonstrating the bug in Safari (Chrome looks similar)

Notice that after time-travelling, clicking the "+" button in the Elm app does nothing, and the user has no way to know why it's not working.

I would expect an overlay to be shown like how Elm 0.18 does it:
<img width="592" alt="elm18 click to resume" src="https://user-images.githubusercontent.com/84525/44312644-50cd3f00-a3b0-11e8-98b0-a45b7d4973a2.png">

**Comment by jinjor** [comment]
> But in 0.19, the overlay is not shown.

It is shown outside of the window. I could make it shown by adding `position: fixed` to the overlay (from devtool), but the "Resume" event was still not handled.

I think this is the event handler but seems like not working.
https://github.com/elm/browser/blob/813607120cad18bae101c9ec502733d9d940c612/src/Debugger/Overlay.elm#L147

**Comment by Pilatch** [comment]
Can we get some action on this issue? I want to demonstrate the awesomeness of Elm and use 0.19, but this regression would make me look silly if I try to show off the debugger.

Sounds like you know what to do, @jinjor

**Comment by jinjor** [comment]
@Pilatch 
I tried debugging hard, but it was really hard (against my expectation) and finally gave up. I think something is wrong in handling virtual DOM in the kernel logic. If you want to debug, you can do something like [this](https://github.com/jinjor/elm-kernel-debug).

**Comment by Pilatch** [comment]
That's pretty cool. I'm seeing a runtime error getting tossed. 

Uncaught TypeError: domNode.replaceData is not a function

`function _VirtualDom_applyPatch(domNode, patch)
{
	switch (patch.$)
	{
		case 0:
			return _VirtualDom_applyPatchRedraw(domNode, patch.s, patch.u);

		case 4:
			_VirtualDom_applyFacts(domNode, patch.u, patch.s);
			return domNode;

		case 3:
			domNode.replaceData(0, domNode.length, patch.s);
			return domNode;`

**Comment by Pilatch** [comment]
But... it looks like this should have been resolved a while back. Maybe it's a regression. https://github.com/elm/virtual-dom/pull/20

**Comment by Pilatch** [comment]
Sorry, I get a very different behavior because I have a page with two Elm apps in it, and it barfs in special ways when I open both debuggers.

**Comment by Pilatch** [comment]
@evancz How do I debug codez in here so I can contribute? I tried jinjor's suggestion of using elm-kernel-debug, but that didn't honor the changes I made in the Browser module and compile them into index.html. 

How do you do it when working on stuff?

**Comment by jinjor** [comment]
@Pilatch Did you understand what the shell script does? It is just a dirty hack, not a library.

**Comment by Pilatch** [comment]
Yea, I understand what it's doing. I wanted to test the browser library, so I made changes in there, put it in the base of the project alongside virtual-dom, changed the `package` variable in the shell script and re-built. The script puts my changes into the 0.19.0 folder, but the generated index.html file doesn't have those changes in it.

**Comment by Pilatch** [comment]
OK, I think I see what's up. Thanks to a helpful person in slack, I discovered that you can negate the cache by deleting .dat files, and by putting stuff in your `~/.elm` folder, which would mean the build script would change line 23 to

`target_dir=~/elm/0.19.0/package/$author/$package/$version`

**Comment by Pilatch** [comment]
Nope. Finally found it. There was a version conflict with virtual-dom. The one I was modifying was 1.0.2, but elm was building 1.0.0. Will update your readme and send pull request... among other things. :)

**Comment by Pilatch** [comment]
Update. Now that I have that running smoothly I can debug enough to confidently say:

1. The overlay can display properly by adding styles in Overlay.elm as @jinjor suggested.

```
                    , style "position" "fixed"
                    , style "top" "0"
                    , style "left" "0"
```

2. Clicking to resume is bugged not because of Elm code, but because of Debugger.js. I was reminded by the adage, "When you find a bug, it's probably in JS." We also know the Elm code is not at fault because the Resume button on the little popout thingy works as expected, and that just sends a simple `Resume` message.

We can prove this by allowing click events through. in the `_Debugger_blocker` function in Debugger.js we can add this check against click events, which will make the overlay's click event work as intended. 

```
	if (event.type !== 'click')
	{
		event.stopPropagation();
		event.preventDefault();
	}
```

Of course that lets all click events through, which will have undesirable consequences and will probably corrupt the program's state in some cases. So my task is now to only allow click events on the overlay. How to do that is not immediately obvious because _hoo-boy_ this code is funky!

**Comment by Pilatch** [comment]
Derp. Looks like I just needed to add `id "elm-debugger-overlay"` to the overlay in Elm. Incoming merge request.

**Comment by Pilatch** [comment]
Created. https://github.com/elm/browser/pull/40

**Comment by evancz** [comment]
The fix in #40 ended up in some debugger improvements that @Skinney worked on recently. The plan is to release that in a PATCH change by Monday, so it should be possible to get the fixed version by just changing from 1.0.1 to 1.0.2 once it is out!
